lib/omniauth/strategies/bitbucket.rb

```diff
@@ -21,18 +22,22 @@ class Bitbucket < OmniAuth::Strategies::OAuth
 
       info do
         {
-          :name => "#{raw_info['first_name']} #{raw_info['last_name']}",
-          :avatar => raw_info['avatar'],
+          :name => raw_info['nickname'],
+          :avatar => raw_info['links']['avatar']['href'],
           :email => raw_info['email']
         }
       end
 
       def raw_info
         @raw_info ||= begin
-                        ri = MultiJson.decode(access_token.get('/api/1.0/user').body)['user']
-                        emails = MultiJson.decode(access_token.get('/api/1.0/emails').body)
-                        email_hash = emails.find { |email| email['primary'] } || emails.first || {}
-                        ri.merge('email' => email_hash['email'])
+                        ri = ::MultiJson.decode(access_token.get('/api/2.0/user').body)
```

Comment:
One question I have is whether this implementation is used for bitbucket self hosted server because if that's the case then this would be a problem

Reply:
No idea, I think the ppl best equipped to answer that have left